src/main/java/webserver/http/request/HttpRequest.java

```diff
@@ -46,11 +49,18 @@ public String getParameter(String key) {
     }
 
     @Override
-    public void process(BufferedReader br) throws IOException {
+    public HttpMethod getMethod() {
+        return requestLine.getHttpMethod();
+    }
+
+    @Override
+    public void process(InputStream in) throws IOException {
+        BufferedReader br = new BufferedReader(new InputStreamReader(in, StandardCharsets.UTF_8));
+
         String line = br.readLine();
         requestLine = RequestLine.parse(line);
-
-        while (!(line = br.readLine()).isEmpty()) {
+        System.out.println(line);
+        while ((line = br.readLine()) != null && !line.isEmpty()) {
```

Comment:
```suggestion
        while (!Strings.isNullOrEmpty(line)) {
```

guava 등의 api를 활용하여 해결할 수도 있습니다.